### PR TITLE
Update tspconfig for data-plane

### LINKED ISSUE #4787: [openapi-to-tsp] Improve tspconfig.yaml
Can we emit a tspconfig.yaml with the following configuration for typespec-autorest: 

```
"@azure-tools/typespec-autorest":
    azure-resource-provider-folder: "data-plane"
    emitter-output-dir: "{project-root}/.."
    examples-directory: "examples"
    output-file: "{azure-resource-provider-folder}/{service-name}/{version-status}/{version}/ContosoWidget.json"
```

This config is the default necessary for checking in specs in the azure rest api specs repo. Maybe this needs a flag to specify that we do this only for Azure specs?

### PR DESCRIPTION
**Pull request by joheredi**
Update data-plane tspconfig to comply with the azure-rest-spec repo requirements

fixes #4787 4787

### CODE REVIEW

File: packages/extensions/openapi-to-typespec/test/anomalyDetector/tsp-output/tspconfig.yaml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 emit:
-  - "@azure-tools/typespec-autorest"
+  - "@azure-tools/typespec-autorest":
```
[the remark this answers is not in the document]
Comment (joheredi) [reply packages/extensions/openapi-to-typespec/test/anomalyDetector/tsp-output/tspconfig.yaml:2]: good catch! Thanks :)

File: packages/extensions/openapi-to-typespec/src/emiters/emit-typespec-config.ts [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
-export async function emitTypespecConfig(filePath: string): Promise<void> {
+export async function emitTypespecConfig(filePath: string, programDetails: TypespecProgram): Promise<void> {
   const session = getSession();
   const { isArm } = getOptions();
   let content = `emit:
-  - "@azure-tools/typespec-autorest"
+  - "@azure-tools/typespec-autorest":
+    azure-resource-provider-folder: "data-plane"
+    emitter-output-dir: "{project-root}/.."
+    examples-directory: "examples"
```
Comment (catalinaperalta) [inline packages/extensions/openapi-to-typespec/src/emiters/emit-typespec-config.ts:14]: Same feedback as Tim about where this should go. 
Also, I think this line should be:
```suggestion
    examples-directory: "{project-root}/examples"
```

File: packages/extensions/openapi-to-typespec/src/emiters/emit-typespec-config.ts [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
+    output-file: "{azure-resource-provider-folder}/{service-name}/{version-status}/{version}/${pascalCase(
+      programDetails.serviceInformation.name,
+    )}.json"
```
Comment (catalinaperalta) [inline packages/extensions/openapi-to-typespec/src/emiters/emit-typespec-config.ts:17]: Do we want to just simplify this and leave it as openapi.json?

Comment (joheredi) [reply packages/extensions/openapi-to-typespec/src/emiters/emit-typespec-config.ts]: Makes sense I like it better

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**